src/metadata-builder/JunctionEntityMetadataBuilder.ts

```diff
@@ -144,20 +145,20 @@ export class JunctionEntityMetadataBuilder {
                           )
                       })
                     : undefined
-                const columnName =
-                    joinColumn && joinColumn.name
-                        ? joinColumn.name
-                        : this.connection.namingStrategy.joinTableInverseColumnName(
-                              relation.inverseEntityMetadata
-                                  .tableNameWithoutPrefix,
-                              inverseReferencedColumn.propertyName,
-                              inverseReferencedColumn.databaseName,
-                          )
+                const isExplicitlyNamed = !!(joinColumn && joinColumn.name)
```

Comment:
I think we don't have the `stylistic-type-checked` ESLint rules enabled yet, but for when we will:

`joinColumn && joinColumn.name` -> `joinColumn?.name`
`!!expression` -> `Boolean(expression)`